src/components/CreationProgress/index.js

```diff
@@ -173,21 +182,23 @@ const CreationProgress = ({
 
         const renderStep = () => {
           if (temporal !== 'future') {
-            return isStepsNumber ? (
-              <Typography color="white" fontWeight={500}>
-                {index + 1}
-              </Typography>
-            ) : (
-              <Icon name="check" color="white" size={20} />
-            )
+            if (animated) {
+              return <Icon name="check" color="white" size={sizes[size].text} />
+            }
+            if (temporal === 'past') {
+              return <Icon name="check" color="white" size={sizes[size].text} />
+            }
```

Comment:
Seems like you are making condition to render exact same component with same properties